plugins/uefi-mok/fu-uefi-mok-common.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-uefi-mok-common.h"
+#include "fu-uefi-mok-struct.h"
+
+FwupdSecurityAttr *
+fu_uefi_mok_attr_new(FuPlugin *plugin, const gchar *filename, GError **error)
+{
+	FuUefiMokHsiKey key_all = FU_UEFI_MOK_HSI_KEY_NONE;
+	g_auto(GStrv) lines = NULL;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GBytes) blob = NULL;
+
+	/* create attr */
+	attr = fu_plugin_security_attr_new(plugin, FWUPD_SECURITY_ATTR_ID_UEFI_MEMORY_PROTECTION);
+	fwupd_security_attr_set_plugin(attr, "uefi_mok");
+	fwupd_security_attr_set_result_success(attr, FWUPD_SECURITY_ATTR_RESULT_LOCKED);
+	fwupd_security_attr_set_level(attr, FWUPD_SECURITY_ATTR_LEVEL_IMPORTANT);
+
+	/* split into lines */
+	blob = fu_bytes_get_contents(filename, error);
+	if (blob == NULL)
+		return NULL;
+	lines = fu_strsplit_bytes(blob, "\n", -1);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		g_auto(GStrv) kv = NULL;
+
+		if (lines[i][0] == '\0')
+			continue;
+		kv = g_strsplit(lines[i], ": ", -1);
```

Comment:
I think you need another check here how many `kv` you got.  If it's not enough; you should `continue;`

Reply:
Ack, added.